src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,291 @@
+		if (!_.isEmpty(event.target.value)) {
+			onExpand({event, props: this.props});
+		} else {
+			onCollapse();
+		}
+	},
+
+	getInputValue() {
+		return _.get(this.refs, 'inputNode.value', this.props.value);
+	},
+
+	setInputValue(value) {
+		if (this.refs.inputNode) {
+			this.refs.inputNode.value = value;
+		}
+	},
+
+	handleInputKeydown(event) {
+		const {
+			onExpand,
+			DropMenu: {
+				isExpanded,
+				focusedIndex,
+				onCollapse,
+			}
+		} = this.props;
+
+		const value = this.getInputValue();
+
+		if (event.keyCode === KEYCODE.Tab && isExpanded && focusedIndex !== null) {
+			this.handleSelect(focusedIndex, {event, props: this.props});
+			event.preventDefault();
+		}
+
+		if (event.keyCode === KEYCODE.Space) {
+			event.stopPropagation();
+		}
+
+		if (event.keyCode === KEYCODE.ArrowDown && !isExpanded) {
+			event.stopPropagation();
+
+			if (_.isEmpty(value)) {
+				onExpand({event, props: this.props});
+			}
+		}
+
+		if (event.keyCode === KEYCODE.Escape) {
+			event.stopPropagation();
+			onCollapse(event);
+		}
+
+		if (event.keyCode === KEYCODE.Enter && focusedIndex === null) {
+			event.stopPropagation();
+			onCollapse(event);
+		}
+	},
+
+	handleControlClick(event) {
+		const {
+			onExpand,
+			DropMenu: {
+				isExpanded,
+				onCollapse
+			}
+		} = this.props;
+
+		if (event.target === this.refs.inputNode) {
+			onExpand({event, props: this.props});
+		} else {
+			if (isExpanded) {
+				onCollapse(event);
+			} else {
+				onExpand({event, props: this.props});
+			}
+
+			this.refs.inputNode.focus();
+		}
+	},
+
+	componentDidMount() {
+		const { value } = this.props;
+		this.refs.inputNode.addEventListener('input', this.handleInput);
+		this.setInputValue(value);
+	},
+
+	componentWillReceiveProps(nextProps) {
+		const { value } = nextProps;
+		if (value !== this.getInputValue()) {
+			this.setInputValue(value);
+		}
+	},
+
+	componentWillUnmount() {
+		if (this.refs.inputNode) {
+			this.refs.inputNode.removeEventListener('input', this.handleInput);
+		}
+	},
+
+	render() {
+		const {
+			style,
+			className,
+			isDisabled,
+			DropMenu: dropMenuProps,
+			suggestions,
+			...passThroughs
+		} = this.props;
+
+		const {
+			direction,
+			isExpanded
+		} = dropMenuProps;
+
+		const value = this.getInputValue();
+		const valuePattern = new RegExp(_.escapeRegExp(value), 'i');
+
+		return (
+			<DropMenu
+				{...dropMenuProps}
+				isDisabled={isDisabled}
+				selectedIndices={[]}
+				className={boundClassNames('&', className)}
+				onSelect={this.handleSelect}
+				style={style}
+			>
+				<DropMenu.Control onClick={this.handleControlClick}>
+					<div className={boundClassNames('&-Control', {
+						'&-Control-is-expanded': isExpanded,
+						'&-Control-is-disabled': isDisabled
+					})}>
+						<input
+							{..._.omit(passThroughs, ['onChange', 'onSelect', 'onExpand', 'value', 'children'])}
+							type='text'
+							className={boundClassNames('&-Control-input')}
+							ref='inputNode'
```

Comment:
i originally had a function, but i changed to a string so that it wouldn't be invoked more than it needed to be, since i just need to keep a reference to the DOMNode once anyways. I don't mind changing it back later.